Testscripts/Linux/containers_utils.sh

```diff
@@ -0,0 +1,219 @@
+HandleSkip()
+{
+    local err_msg=$1
+    local err_code=$2
+
+    SetTestStateSkipped
+    LogErr "$err_msg (ret: $err_code)"
+    exit 0
+}
+
+# Function to verify docker engine is started correctly
+function VerifyDockerEngine()
+{
+    LogMsg "VerifyDockerEngine on $DISTRO"
+
+    output=$(docker run hello-world)
+    if [ $? -ne 0 ]; then
+        LogErr "Fail to run docker run hello-world"
+        return 1
+    fi
+    LogMsg "VerifyDockerEngine: hello-world output - $output"
+}
+
+# Function to start docker engine
+function StartDockerEngine()
+{
+    LogMsg "Start docker engine"
+    systemctl start docker || service docker start
+    if [ $? -ne 0 ]; then
+        LogErr "Fail to start docker service."
+        return 1
+    fi
+}
+
+# Function to install docker engine
+function InstallDockerEngine()
+{
+    local ret=0
+
+    LogMsg "InstallDockerEngine on $DISTRO"
+    update_repos
+
+    case $DISTRO in
+        ubuntu*|debian*)
+            LogMsg "Uninstall old versions of Docker."
+            apt-get remove -y docker docker-engine docker.io containerd runc
+            LogMsg "Install packages apt-transport-https ca-certificates curl gnupg-agent software-properties-common."
+            apt-get update
+            install_package "apt-transport-https ca-certificates curl gnupg-agent software-properties-common"
+            LogMsg "Add Docker's official GPG key."
+            curl -fsSL https://download.docker.com/linux/$DISTRO_NAME/gpg | sudo apt-key add -
+            LogMsg "Set up the stable repository."
+            add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/$DISTRO_NAME $(lsb_release -cs) stable"
+            apt-get update
+            install_package "docker-ce docker-ce-cli containerd.io"
+            ret=$?
+        ;;
+
+        centos*|redhat*)
+            LogMsg "Uninstall old versions of Docker."
+            yum remove -y docker docker-client docker-client-latest docker-common docker-latest docker-latest-logrotate docker-logrotate docker-engine
+            LogMsg "Install package yum-utils."
+            install_package "yum-utils"
+            LogMsg "Add repo https://download.docker.com/linux/centos/docker-ce.repo."
+            yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+            if [[ $DISTRO_VERSION == 8* ]];then
+                yum install --nogpgcheck -y docker-ce docker-ce-cli containerd.io --nobest
+            elif [[ $DISTRO_VERSION == 7* ]];then
+                yum install http://mirror.centos.org/centos/7/extras/x86_64/Packages/container-selinux-2.107-1.el7_6.noarch.rpm
+                yum install --nogpgcheck -y docker-ce docker-ce-cli containerd.io
+            else
+                LogMsg "Test not enabled for RH/CentOS 6X"
+            fi
+            ret=$?
+        ;;
+
+    *)
+        LogMsg "$DISTRO not supported"
+        ret=2
+    esac
+    [[ $ret -ne 0 ]] && return $ret
+
+    StartDockerEngine; ret=$?
```

Comment:
StartDockerEngine & VerifyDockerEngine are not used in other places? Do you plan to call them in future? Otherwise, there shouldn't be function.

Reply:
Yes